File: CS/CustomBarSubItem/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using CustomBar.CustomBarManager;
using DevExpress.XtraBars;

namespace CustomBar {
    public partial class Form1 : Form {
        
        public Form1() {
            InitializeComponent();
            CustomBarBuilding();
        }

        public void CustomBarBuilding() {

            BarCheckItem menuCheckItem = new BarCheckItem(customBarManager1, false) { Caption = "customCheckItem"};

            CustomBarSubItem subItem = new CustomBarSubItem(customBarManager1, "Menu");

            BarCheckItem subMenuCheckItem = new BarCheckItem(customBarManager1, false) { Caption = "checkItem1"};
            BarCheckItem subMenuCheckItem1 = new BarCheckItem(customBarManager1, false) { Caption = "checkItem2" };
            BarCheckItem subMenuCheckItem2 = new BarCheckItem(customBarManager1, false) { Caption = "checkItem3" };
            
            subItem.AddItems(new BarItem[] { subMenuCheckItem, subMenuCheckItem1, subMenuCheckItem2 });
            bar2.AddItems(new BarItem[] { subItem, menuCheckItem });
        }

    }
}

File: VB/CustomBarSubItem/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Forms
Imports CustomBar.CustomBarManager
Imports DevExpress.XtraBars

Namespace CustomBar
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
			CustomBarBuilding()
		End Sub

		Public Sub CustomBarBuilding()

			Dim menuCheckItem As New BarCheckItem(customBarManager1, False) With {.Caption = "customCheckItem"}

			Dim subItem As New CustomBarSubItem(customBarManager1, "Menu")

			Dim subMenuCheckItem As New BarCheckItem(customBarManager1, False) With {.Caption = "checkItem1"}
			Dim subMenuCheckItem1 As New BarCheckItem(customBarManager1, False) With {.Caption = "checkItem2"}
			Dim subMenuCheckItem2 As New BarCheckItem(customBarManager1, False) With {.Caption = "checkItem3"}

			subItem.AddItems(New BarItem() { subMenuCheckItem, subMenuCheckItem1, subMenuCheckItem2 })
			bar2.AddItems(New BarItem() { subItem, menuCheckItem })
		End Sub

	End Class
End Namespace
